ocaml-lsp-server/src/workspace_symbol.ml

```diff
@@ -0,0 +1,261 @@
+open Import
+
+let to_loc (loc : Warnings.loc) : Loc.t =
+  { loc_start = loc.loc_start
+  ; loc_end = loc.loc_end
+  ; loc_ghost = loc.loc_ghost
+  }
+
+let cmt_sign (cmt_infos : Cmt_format.cmt_infos) =
+  match cmt_infos.cmt_annots with
+  | Cmt_format.Implementation { Typedtree.str_type = sign; _ }
+  | Cmt_format.Interface { Typedtree.sig_type = sign; _ }
+  | Cmt_format.Packed (sign, _) ->
+    sign
+  | _ -> []
+
+let loc_of_sig_item (signature_item : Types.signature_item) =
+  match signature_item with
+  | Types.Sig_value (_, descr, _) -> descr.val_loc
+  | Types.Sig_type (_, descr, _, _) -> descr.type_loc
+  | Types.Sig_typext (_, descr, _, _) -> descr.ext_loc
+  | Types.Sig_module (_, _, descr, _, _) -> descr.md_loc
+  | Types.Sig_modtype (_, descr, _) -> descr.mtd_loc
+  | Types.Sig_class (_, descr, _, _) -> descr.cty_loc
+  | Types.Sig_class_type (_, descr, _, _) -> descr.clty_loc
+
+let id_of_sig_item (signature_item : Types.signature_item) =
+  match signature_item with
+  | Types.Sig_value (id, _, _)
+  | Types.Sig_type (id, _, _, _)
+  | Types.Sig_typext (id, _, _, _)
+  | Types.Sig_module (id, _, _, _, _)
+  | Types.Sig_modtype (id, _, _)
+  | Types.Sig_class (id, _, _, _)
+  | Types.Sig_class_type (id, _, _, _) ->
+    id
+
+let kind_of_sig_item (signature_item : Types.signature_item) : SymbolKind.t =
+  match signature_item with
+  | Types.Sig_value _ -> Function
+  | Types.Sig_type _ -> String
+  (* | Types.Sig_typext (_, _, Types.Text_exception, _) -> Constructor *)
+  | Types.Sig_typext _ -> Constructor
+  | Types.Sig_module _ -> Module
+  | Types.Sig_modtype _ -> Module
+  | Types.Sig_class _ -> Class
+  | Types.Sig_class_type _ -> Class
+
+let symbol_of_signature_item ~sourcefile (signature_item : Types.signature_item)
+    =
+  let id = id_of_sig_item signature_item in
+  let loc = loc_of_sig_item signature_item in
+  let name = Ident.name id in
+  let kind = kind_of_sig_item signature_item in
+  let range = Range.of_loc (loc |> to_loc) in
+  let uri = sourcefile |> Uri.of_path in
+  let location = Location.create ~uri ~range in
+  SymbolInformation.create ~name ~kind ~location ()
+
+let build_roots =
+  (* by increasing order of priority *)
+  [ "_darcs"; ".hg"; ".git"; "jengaroot.ml"; "omakeroot"; "_build"; "_obuild" ]
```

Comment:
Let's just stick to `_build`. There's no need for half hearted support for other build systems. If we have users interested in using other build systems, we'll think of a proper way to integrate.

Reply:
Yes, that makes sense. This code was copied from `ocp-index` and I didn't want to modify it before I had something that works.